src/utils/wallet.js

```diff
@@ -18,15 +20,28 @@ export function WalletProvider({ children }) {
     'https://www.sollet.io',
   );
 
-  const wallet = useMemo(() => new Wallet(providerUrl, endpoint), [
+  let wallet = useMemo(() => new Wallet(providerUrl, endpoint), [
     providerUrl,
     endpoint,
   ]);
+  const [ezWallet] = useState(new EZWallet());
```

Comment:
put EZWallet initialization code in the const wallet = useMemo(...)

Reply:
something like this
```
const wallet = useMemo(() => {
  if (new URL(providerUrl).hostname === 'ezdefi.com') {
    return new EzWallet();
  }
  return new Wallet(providerUrl, endpoint);
}, [
    providerUrl,
    endpoint,
  ]);
```

Reply:
Tried it but it works incorrectly.
The default wallet was initialization is sollet.io. When switching the wallet, the wallet object is not re-initialization, an error: "wallet.on is not a function".